static/app/views/issueList/customViewsHeader.tsx

```diff
@@ -257,20 +259,49 @@ function CustomViewsIssueListHeaderTabsContent({
               tab.id = view.id;
             }
           });
-          navigate({
-            ...location,
-            query: {
-              ...queryParams,
-              viewId: tab.id,
+          navigate(
+            {
+              ...location,
+              query: {
+                ...queryParams,
+                viewId: tab.id,
+              },
             },
-          });
+            {replace: true}
+          );
         }
         return tab;
       })
     );
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [views]);
 
+  useEffect(() => {
+    if (viewId?.startsWith('_')) {
+      // If the user types in query manually while the new view flow is showing,
+      // then replace the add view flow with the issue stream with the query loaded,
+      // and persist the query
+      if (newViewActive && query !== '') {
+        setNewViewActive(false);
+        const updatedTabs: Tab[] = draggableTabs.map(tab =>
+          tab.id === viewId
+            ? {
+                ...tab,
+                unsavedChanges: [query, sort ?? IssueSortOptions.DATE],
+              }
+            : tab
+        );
+        setDraggableTabs(updatedTabs);
+        debounceUpdateViews(updatedTabs);
+      } else {
+        setNewViewActive(true);
+      }
+    } else {
+      setNewViewActive(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [viewId, query]);
```

Comment:
This ensures that the add view flow only appears on new views that have not been saved to the backend yet. It also ensures that if a user decides to skip the add view flow and write their own query, the query will be saved to the tab afterwards.